program/src/processor.rs

```diff
@@ -125,16 +131,18 @@ pub fn process_initialize(
             developer_account: *accounts.developer_account.key,
         },
         metrics: Metrics::new(),
-        maintainers: Maintainers::new(max_maintainers),
-        validators: Validators::new(max_validators),
         max_commission_percentage,
     };
 
     // Confirm that the fee recipients are actually stSOL accounts.
     lido.check_is_st_sol_account(accounts.treasury_account)?;
     lido.check_is_st_sol_account(accounts.developer_account)?;
 
-    lido.save(accounts.lido)
+    // first save Lido, then maintainer list and then validator list
+    let lido_data = &mut *accounts.lido.data.borrow_mut();
+    lido.save(lido_data)?;
+    maintainers.save(&mut lido_data[LIDO_CONSTANT_SIZE..])?;
```

Comment:
What does it need to save here? The `ValidatorList` and `MaintainerList` mutate the backing account directly, so there is nothing to save, right?

Reply:
Yes. But we have to serialize empty `AccountList`'s once on initialization. From now on we use BigVec only. We use `borsh` here so that we do not have to implement manually BigVec initial serialization: serializing `ListHeader` struct and putting number of elements in `BigVec` ahead. So `borsh` helps us to init. That's John's idea, pretty good